core/arch/aarchxx/mangle.c

```diff
@@ -665,6 +665,8 @@ insert_push_all_registers(dcontext_t *dcontext, clean_call_info_t *cci,
     dstack_offs += 15 * XSP_SZ;
 
     /* Make dstack_offs 8-byte algined, as we only accounted for 17 4-byte slots. */
+    PRE(ilist, instr,
```

Comment:
Assuming there's an odd number of XSP_SZ-sized writes to the stack and that's why we need this fix: I was trying to locate the XSP_SZ-sized stack write with the missing parity. Is it the one on L655? But that happens only for DR_ISA_ARM_THUMB isn't it?

Maybe a comment clarifying the accounting would help.